pkg/gui/context/working_tree_context.go

```diff
@@ -32,6 +33,7 @@ func NewWorkingTreeContext(c *ContextCommon) *WorkingTreeContext {
 
 	ctx := &WorkingTreeContext{
 		SearchTrait:       NewSearchTrait(c),
+		HistoryTrait:      NewHistoryTrait(c),
```

Comment:
Is it possible to embed the history trait inside the search trait? (Likewise with the filter trait)? If we could do that then it would be a lot nicer and more encapsulated

Reply:
I embedded the `HistoryTrait` in the `SearchTrait`. But I can't seem to find a `FilterTrait`. Closest thing I could find was a `ListContextTrait`, but I am not sure that's what we're looking for here.

changes to `SearchTrait` are here: 8e73ffe7d5f43b0b51a293c0c3096f93d5966392

Reply:
Nice. I just pushed a change to put it inside FilteredListViewModel (also did some renaming)

Reply:
oh, good call. think I got too hung up on the names.